### PR TITLE
[Bugfix] JS: Axis orientation projection detection

### PR DESCRIPTION
**Pull request by rldhont**
Re-implementation of the projection axis orientation detection method.

Funded by VSB

### CODE REVIEW

File: assets/src/modules/Lizmap.js [hunk 1]
```diff
@@ -46,19 +47,66 @@ export default class Lizmap {
...
                         proj4.defs(ref, def);
                     }
                 }
-
+                // Register project projection if unknown
+                const configProj = this._initialConfig.options.projection;
+                if (configProj.ref !== "" && !proj4.defs(configProj.ref)) {
+                    proj4.defs(configProj.ref, configProj.proj4);
+                }
+                // Add CRS:84 projection, same as EPSG:4326 but with enu axis orientation
```
Comment (Gustry) [inline assets/src/modules/Lizmap.js:62]:
```suggestion
                // Add CRS:84 projection, same as EPSG:4326 but with ENU axis orientation
```

File: assets/src/modules/Lizmap.js [hunk 2]
```diff
@@ -46,19 +47,66 @@ export default class Lizmap {
...
+                proj4.defs("CRS:84","+proj=longlat +datum=WGS84 +no_defs +type=crs");
                 register(proj4);
+                // Update project projection if is axis orientation is not enu
```
Comment (Gustry) [inline assets/src/modules/Lizmap.js:65]:
```suggestion
                // Update project projection if its axis orientation is not ENU
```

File: assets/src/modules/Lizmap.js [hunk 3]
```diff
@@ -46,19 +47,66 @@ export default class Lizmap {
...
+                    // loop through bounding boxes of the project provided by WMS capabilities
+                    for (const bbox of wmsCapabilities.Capability.Layer.BoundingBox) {
+                        // If the BBOX CRS is not the same of the project projection, continue.
+                        if (bbox.crs !== configProj.ref) {
+                            continue;
+                        }
+                        // Get project projection
+                        const projectProj = getProjection(configProj.ref);
+                        console.log(projectProj);
+                        // Check axis orientation, if it is not enu, break, we don't have to do anything
```
Comment (Gustry) [inline assets/src/modules/Lizmap.js:76]:
```suggestion
                        // Check axis orientation, if it is not ENU, break, we don't have to do anything
```

File: tests/end2end/playwright/axis_orientation.spec.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
+// @ts-check
+const { test, expect } = require('@playwright/test');
+
+test.describe('Axis Orientation', () => {
+
+    test('Axis Orientation neu for EPSG:3044', async ({ page }) => {
```
Comment (Gustry) [inline tests/end2end/playwright/axis_orientation.spec.js:6]:
```suggestion
    test('Axis Orientation NEU for EPSG:3044', async ({ page }) => {
```

File: tests/end2end/playwright/axis_orientation.spec.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        expect(GetTiles[0]).toContain('6/33/20.png')
+        expect(GetTiles[1]).toContain('6/33/21.png')
+        expect(GetTiles[2]).toContain('6/34/20.png')
+        expect(GetTiles[3]).toContain('6/34/21.png')
+        expect(GetTiles[4]).toContain('6/33/22.png')
+        expect(GetTiles[5]).toContain('6/34/22.png')
+        await page.unroute('https://tile.openstreetmap.org/*/*/*.png')
+    });
+
+    test('Axis Orientation neu for EPSG:3844', async ({ page }) => {
```
Comment (Gustry) [inline tests/end2end/playwright/axis_orientation.spec.js:54]:
```suggestion
    test('Axis Orientation NEU for EPSG:3844', async ({ page }) => {
```

**Review by Gustry — APPROVED** [review]
Minor tweaks, looks easier to understand when it's an acronym

File: assets/src/modules/Lizmap.js (hunk 1 above)
Comment (Gustry) [reply assets/src/modules/Lizmap.js:62]: It helps with a link I think about this acronym, what do you think ?

**Review by nboisteault — APPROVED** [review]

File: assets/src/modules/Lizmap.js [hunk 6]
```diff
@@ -46,19 +47,66 @@ export default class Lizmap {
...
+                if (configProj.ref !== "") {
+                    // loop through bounding boxes of the project provided by WMS capabilities
+                    for (const bbox of wmsCapabilities.Capability.Layer.BoundingBox) {
+                        // If the BBOX CRS is not the same of the project projection, continue.
+                        if (bbox.crs !== configProj.ref) {
+                            continue;
+                        }
+                        // Get project projection
+                        const projectProj = getProjection(configProj.ref);
+                        console.log(projectProj);
```
Comment (rldhont) [inline assets/src/modules/Lizmap.js:75]:
```suggestion
```

**Pull request merged**